ecommerce/enterprise_coupons/views.py

```diff
@@ -0,0 +1,54 @@
+"""
+Views for enterprise coupons application.
+"""
+
+import logging
+
+from django.http import HttpResponse
+from django.shortcuts import redirect
+from django.views import View
+
+from ecommerce.enterprise_coupons.backends import get_enterprise_catalog_uuid_from_coupon
+from ecommerce.enterprise_coupons.exceptions import CatalogueNotFoundError, CouponError, CouponNotFoundError
+from ecommerce.enterprise_coupons.utils import get_build_mfe_base_url
+
+logger = logging.getLogger(__name__)
+
+
+class CouponRedirectView(View):
+    """
+    View to redirect users to the MFE catalog page based on a coupon code.
+
+    This view extracts the catalog UUID from the provided coupon code and redirects
+    the user to the appropriate MFE catalog page with the catalog UUID as a parameter.
+
+    Returns:
+        - 302: Redirect to MFE catalog page on success
+        - 404: Empty response for coupon or catalogue not found
+        - 400: Empty response for invalid coupon
+        - 500: Empty response for configuration issues
+
+    Note:
+        All errors are logged internally with detailed information for debugging,
+        but NO messages are returned to clients to avoid exposing implementation
+        details or providing information to potential attackers.
+    """
+    def get(self, request, coupon_code, *args, **kwargs):  # pylint: disable=unused-argument
+        try:
+            catalog_uuid = get_enterprise_catalog_uuid_from_coupon(coupon_code)
+            base_url = request.site.siteconfiguration.lms_url_root
+            mfe_url = get_build_mfe_base_url(catalog_uuid, base_url)
+
+            return redirect(mfe_url)
+        except CouponNotFoundError as e:
+            logger.info('Coupon not found: %s - Error: %s', coupon_code, str(e))
```

Comment:
```suggestion
            logger.error('Coupon not found: %s - Error: %s', coupon_code, str(e))
```
I think this could be an error as well. @daviidco

Reply:
Done